contents/handbook/engineering/release-new-version.md

```diff
@@ -14,49 +14,55 @@ Hopefully we will not have to do many patch versions, but if between versions we
-  - `git add CHANGELOG.md`
-  - `git commit -m "Changelog version 1.7.0"`
-
-<br />
+Three days before we release, on Thursday, we institute a code freeze. We branch master into release-[version] and deploy that to our playground environment (playground.posthog.com). Only bugfixes are allowed to be merged into this branch (and thus put on production) between the code freeze and the release going out. This gives us about three days to test if this release has any bugs.
 
+<blockquote class="warning-note">
+⚠️ As soon as the branch is created and pushed to GitHub, the Docker image will be built and pushed to Docker Hub under the tag <code>release-[version]</code>.
+</blockquote>
 
-<input type="checkbox"/> Update the `VERSION` in `posthog/version.py`
-  - `git checkout release-[version]`
-  - `git add posthog/version.py`
-  - `git commit -m "Bump version [version]"`
-
-<br />
-
-<input type="checkbox"/> Tag the version
-  - `git tag -a [version] -m "Version [version]"`
-  - `git push origin head --tags`
-
+## Checklist
 
-Once a new Docker image has been built (see [Docker Hub](https://hub.docker.com), password in 1Password) for the new version, open the [charts](https://github.com/PostHog/charts) repo and make the changes:
 
-1. Edit the **two** Chart files: [Chart.yaml](https://github.com/PostHog/charts/blob/master/charts/posthog/Chart.yaml) and [ChartV3.yaml](https://github.com/PostHog/charts/blob/master/charts/posthog/ChartV3.yaml), in both: 
+### Initial setup & code freeze phase
+
+- [ ] Start the `release-[version]` branch to initiate the code freeze.
+- [ ] Figure out what's updated in this release with the command below. The command will output the entire commit list to `log.txt`. You can use this list to obtain external contributions to highlight in the Array.
+  ```bash
+  git checkout release-[version]
+  git log --pretty=format:"%s %ae" [old-version]..head > log.txt
+  ```
+- [ ] Write up the main fixes/improvements and breaking changes into `CHANGELOG.md` following the structure of the previous release
+  ```bash
+  git add CHANGELOG.md
+  git commit -m "Changelog version 1.7.0"
+  ```
+- [ ] Update the `VERSION` in `posthog/version.py`
+  ```bash
+  git checkout release-[version]
+  git add posthog/version.py
+  git commit -m "Bump version [version]"
+  ```
+- [ ] Deploy PostHog playground from the `release-[version]` branch on Heroku dashboard.
+- [ ] **Break the release session!** It's imperative that the session uses the published `release-[version]` branch from Docker Hub (this is published automatically using GitHub Actions), to avoid any potential bugs creeping up in the final build stage.
+
+### Launch phase
+- [ ] Write up the [PostHog Array blog post](/handbook/growth/marketing/blog#posthog-array)
+- [ ] Edit the **two** Chart files: [Chart.yaml](https://github.com/PostHog/charts/blob/master/charts/posthog/Chart.yaml) and [ChartV3.yaml](https://github.com/PostHog/charts/blob/master/charts/posthog/ChartV3.yaml), in both: 
```

Comment:
@fuziontech @tiina303 we now have the `charts-clickhouse` repo, should we update both repos or just one and which one?